tests/k6/scenarios/core2_qa_control_plane_ops.js

```diff
@@ -225,29 +247,77 @@ function handleCtlOp(config, op, modelTypeIx, existingModels) {
             }
             break;
 
-        // ignore opOk == seldonOpExecStatus.CONCURRENT_OP_FAIL case
-        // it just means another VU did an operation invalidating the one of the
+        // Ignore opOk == seldonOpExecStatus.CONCURRENT_OP_FAIL case;
+        // It just means another VU did an operation invalidating the one of the
         // current VU, concurrently; example: current VU wants to update a model,
-        // another VU already deleted it
+        // another VU already deleted it. It is expected that such operations
+        // would fail, and they shouldn't be recorded as errors as long as the
+        // control plane remains functional
     }
 
     if (targetCounter != null) {
         targetCounter.add(1)
     }
 
     return opOk
-
 }
 
 export default function (config) {
-    kubeclient = k8s.init()
+    kubeClient = k8s.init()
+    if(config.enableStateCheck) {
+        if (periodicExclusiveRun(config.checkStateEverySec,
+                                config.maxCheckTimeSec,
+                                maxIterDuration, checkStatus)) {
+            console.log(`VU ${vu.idInTest} starts a state consistency check...`)
+            // Perform state consistency checks
+            let k8sModels = k8s.getAllModels()
+
+            if (schedClient === null) {
+                schedClient = scheduler.connectScheduler(config.schedulerEndpoint)
+            }
+
+            // The folowing code all executes asynchronously; It's the only way
+            // we can currently get grpc status streaming from the scheduler.
+            //
+            // The VU will return immediately, but all the async functions below
+            // are registered to run as part of the event loop, and k6 will wait
+            // some time for them to complete before ending the current iteration
+            scheduler.getAllModels().then((schedModels) => {
+                let msc = checkModelsStateIsConsistent(k8sModels, schedModels)
+
+                if (config.isLoadPipeline) {
+                    let k8sPipelines = k8s.getAllPipelines()
+                    scheduler.getAllPipelines().then((schedPipelines) => {
+                        let psc = checkPipelinesStateIsConsistent(k8sPipelines, schedPipelines)
+
+                        if (!psc && config.stopOnCheckFailure) {
+                            test.abort("Aborting test due to pipeline state inconsistencies")
+                        }
+                    })
+                }
+
+                if (!msc && config.stopOnCheckFailure) {
+                    test.abort("Aborting test due to model state inconsistencies")
+                }
+            })
+
+            // required for the async operations set above to run before the next
+            // iteration starts
+            return
+        }
+    }
+
     const numModelTypes = config.modelType.length
 
-    var idx = Math.floor(Math.random() * numModelTypes)
```

Comment:
Rather than picking a random model type and then checking whether we have configured at least 1 model of that type, we now only directly generate a random index amongst the ones that are > 0. Giving up on the rejection sampling strategy means we avoid remaining in the loop for too long (i.e when lots of model types are configured to have a max of 0 models)